portal/src/main/webapp/WEB-INF/jsp/dashboard/dashboard.jsp

```diff
@@ -272,43 +282,73 @@
-					StudyViewParams.params = {
-					    studyId: _cancerStudy.id,
-					    mutationProfileId: _mutationProfiles.length>0?_mutationProfiles[0].id:'',
-					    hasMutSig: hasMutation,
-                        caseSetId: _cancerStudy.id + '_all',
-					    cnaProfileId: _cnaProfiles.length>0?_cnaProfiles[0].id:''
-					};
-					window.mutationProfileId = StudyViewParams.params.mutationProfileId ;
-					window.cnaProfileId = StudyViewParams.params.cnaProfileId;
-					window.case_set_id = StudyViewParams.params.caseSetId;
+
+        var getVirtualStudies = function(){
+        		var _def = new $.Deferred();
+        		if (vcSession.URL !== undefined) {
+                  $.ajax({
+                      method: 'GET',
+                      url: vcSession.URL
+                  }).done(function(response){
+                	  	_def.resolve(response)
+                  });
+            } else {
+            		_def.resolve([])
+			}
+        		return _def.promise();
+        }
+        $.when(window.cbioportal_client.getStudies({ study_ids: studyIds}), window.iviz.datamanager.getGeneticProfiles(),getVirtualStudies())
+            .then(function(_cancerStudies, _geneticProfiles,virtualStudies){
+				if(cohortIdsList.length === 1 ) {
+					if(JSON.stringify(cohortIdsList) === JSON.stringify(studyIds)) {
+        					$("#show_study_details").css('display', 'block');
+			        		var _cancerStudy = _cancerStudies[0]
+						document.title = _cancerStudy.name
+						$("#study_name").html(_cancerStudy.name);
+						var _desc = _cancerStudy.description;
+						if(_cancerStudy.pmid !== null){
+						    _desc += '&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pubmed/'+_cancerStudy.pmid+'">PubMed</a>';
+						}
+						$("#study_desc").html(_desc);
+						
+						var _mutationProfiles = _.filter(_geneticProfiles, function (_profile) {
+						    return _profile.study_id + '_mutations' === _profile.id;
+						});
+						if(_mutationProfiles.length>0){
+						    appendMutationTab();
+						}
+						var _cnaProfiles = _.filter(_geneticProfiles, function (_profile) {
+						    return _profile.study_id + '_gistic' === _profile.id;
+						});
+						if(_cnaProfiles.length>0){
+						    appendCnaTab();
+						}
+						
+						// TODO changed mutationProfileId to mutationProfileIds when mutations tab support multi-studies
+						StudyViewParams.params = {
+						    studyId: _cancerStudy.id,
+						    mutationProfileId: _mutationProfiles.length>0?_mutationProfiles[0].id:'',
+						    hasMutSig: hasMutation,
+						    caseSetId: _cancerStudy.id + '_all',
+						    cnaProfileId: _cnaProfiles.length>0?_cnaProfiles[0].id:''
+						};
+						window.mutationProfileId = StudyViewParams.params.mutationProfileId ;
+						window.cnaProfileId = StudyViewParams.params.cnaProfileId;
+						window.case_set_id = StudyViewParams.params.caseSetId;
+        				} else {
+        					var response = _.findWhere(virtualStudies, {id: cohortIdsList[0]})
```

Comment:
Just curious, in what situation cohortIdsList may contain no physical id, no virtual id?

Reply:
`cohortIdsList` can't be empty or null. It should have one or more physical id and virtual id

Reply:
Nevermind, I just realize when cohortIdsList only has one element and it is not physical study, not virtual study, then it has to be a combined study, separated by a comma.